src/theoretical.jl

```diff
@@ -64,6 +64,23 @@ Evaluate the variogram at points `x` and `y`.
 #------------------
 # IMPLEMENTATIONS
 #------------------
+"""
+    NuggetEfect(n)
+
+A pure nugget effect (random) variogram with nugget `n`
+"""
+@with_kw struct NuggetEffect{T,D} <: Variogram{T,D}
+  nugget::T = 0.0
+  distance::D = Euclidean()
+end
+
+(γ::NuggetEffect)(h) = (h > 0) * γ.nugget
+Base.range(::NuggetEffect{T,D}) where {T,D} = zero(T)
+sill(γ::NuggetEffect) = γ.nugget
+isstationary(::Type{<:NuggetEffect}) = true
+
+NuggetEffect(n) = NuggetEffect(nugget=n)
+
```

Comment:
Please move this outer constructor to right after the struct definition, I think we don't even need a blank line:

```julia
@with_kw struct NuggetEffect{T,D} <: Variogram{T,D}
  nugget::T = 0.0
  distance::D = Euclidean()
end
NuggetEffect(n) = NuggetEffect(nugget=n)
(γ::NuggetEffect)(h) = (h > 0) * γ.nugget
Base.range(::NuggetEffect{T,D}) where {T,D} = zero(T)
...
```

Reply:
Done!